logprep/processor/domain_resolver/processor.py

```diff
@@ -158,35 +180,49 @@ def _apply_rules(self, event, rule):
             return
         self.metrics.total_urls += 1
         if self._config.cache_enabled:
-            hash_string = self._hasher.hash_str(domain, salt=self._config.hash_salt)
-            requires_storing = self._cache.requires_storing(hash_string)
-            if requires_storing:
-                resolved_ip = self._resolve_ip(domain, hash_string)
-                self._domain_ip_map.update({hash_string: resolved_ip})
-                self.metrics.resolved_new += 1
-            else:
-                resolved_ip = self._domain_ip_map.get(hash_string)
-                self.metrics.resolved_cached += 1
-            self._add_resolve_infos_to_event(event, rule, resolved_ip)
-            if self._config.debug_cache:
-                self._store_debug_infos(event, requires_storing)
+            self._resolve_with_cache(domain, event, rule)
         else:
-            resolved_ip = self._resolve_ip(domain)
+            resolved_ip, _ = self._resolve_ip(domain)
             self._add_resolve_infos_to_event(event, rule, resolved_ip)
 
+    def _resolve_with_cache(self, domain, event, rule):
+        hash_string = self._hasher.hash_str(domain, salt=self._config.hash_salt)
+        requires_storing = self._cache.requires_storing(hash_string)
+        if requires_storing:
+            resolved_ip, status = self._resolve_ip(domain)
+            if status in (ResolveStatus.SUCCESS, ResolveStatus.UNKNOWN, ResolveStatus.TIMEOUT):
+                self._domain_ip_map.update({hash_string: resolved_ip})
+            self.metrics.resolved_new += 1
+        else:
+            resolved_ip = self._domain_ip_map.get(hash_string)
+            self.metrics.resolved_cached += 1
+        self._add_resolve_infos_to_event(event, rule, resolved_ip)
+
+        if self._config.debug_cache:
+            self._store_debug_infos(event, requires_storing)
+
     def _add_resolve_infos_to_event(self, event, rule, resolved_ip):
         if resolved_ip:
             self._write_target_field(event, rule, resolved_ip)
 
     def _resolve_ip(self, domain, hash_string=None):
+        """Resolve domain with timeout.
+
+        Assumes socket default timeout is None and relies on threading to create a timeout.
+        """
         try:
             result = self._thread_pool.apply_async(socket.gethostbyname, (domain,))
             resolved_ip = result.get(timeout=self._config.timeout)
-            return resolved_ip
-        except (context.TimeoutError, OSError):
-            if hash_string:
-                self._domain_ip_map[hash_string] = None
+            return resolved_ip, ResolveStatus.SUCCESS
+        except ValueError:  # Makes no connection so does not need to be cached
+            self.metrics.invalid_domains += 1
+            return None, ResolveStatus.INVALID
+        except context.TimeoutError:
```

Comment:
```suggestion
        except TimeoutError:
```

Why do you use the TimeoutError from "context" ??

Reply:
It's a different error despite having the same name. It's thrown by the context manager and the regular TimoutError can't be caught in its place.